### ISSUE TITLE
storage permission on android 13

### ISSUE THREAD

**Issue by freedom-foundation** [osmdroid/osmdroid#2060]
on android 13 osmdroid 6.1.20 during the intro it is not asking for storage permissions and also for this cannot change locations of data

**Comment by ex4dev** [comment]
For anyone else who had this issue, instead of getting external storage permissions, you can change osmdroid to use your app's internal cache directory:
```kotlin
Configuration.getInstance().osmdroidBasePath = context.cacheDir
```

**Comment by freedom-foundation** [comment]
And also for this into setup cannot change locations of data. I notice the internal sdcard is listed as an option but cannot be selected.


> For anyone else who had this issue, instead of getting external storage permissions, you can change osmdroid to use your app's internal cache directory:
> 
WARNING: If the internal cache dir is inaccesable to the user large cache files of map data may not only use up space but become unusable while using up space. I had this same problem with F-Droid.
 Not so sure this is really cache or that you mean the internal dir is labeled by android for the purpose of cache- it seems we have misused the word "cache".


> ```kotlin
> Configuration.getInstance().osmdroidBasePath = context.cacheDir
> ```

What is this code snippet?: a configuratiin file. Now that I reply I can see kotlin: Is this a build-time configuration?

**Comment by freedom-foundation** [comment]
Hello, can you fix this bug?

**Comment by spyhunter99** [comment]
the configuration object is effectively a configuration file that is specific to the apk. we tried to setup reasonable defaults, but android is a constantly evolving ecosystem...so those defaults might be wrong today.

not sure why it is not asking for permission. you might be able to use the android app info to manually enable the permissions.

long story short, the permissions are used so that we can write a cache on the device of downloaded map tiles. If we dont cache them, the performance is awful and then you're constantly downloading tiles.

**Comment by freedom-foundation** [comment]
Yes, we need to adapt the app to work in the current android ecosystem. This is a good use of branch in git because the old editions can remain. I have the android ecosystem overview we need. It can be found online but if you ask I can make a gh repo. 
> the configuration object is effectively a configuration file that is specific to the apk. we tried to setup reasonable defaults, but android is a constantly evolving ecosystem...so those defaults might be wrong today.

The permission is there but maybe it is the selecting of a subdir which is never prompted. I take it the app will have basic write permission then permission to a single dir. 
> not sure why it is not asking for permission. you might be able to use the android app info to manually enable the permissions.

Yes, it is well to save the map data anyway.
> long story short, the permissions are used so that we can write a cache on the device of downloaded map tiles. If we dont cache them, the performance is awful and then you're constantly downloading tiles.

**Comment by freedom-foundation** [comment]
...a bunch of bad people calling a grown man a baby for nonstop bugreports because they are screwups and keep feeding him trash ( so the hijacking of the code is almost negligable )...

I could see where a guy siezes the code of thousands of developers and can't keep up with the maintinence. What are you doing here? @spyhunter99 
> Yes, we need to adapt the app to work in the current android ecosystem. This is a good use of branch in git because the old editions can remain. I have the android ecosystem overview we need. It can be found online but if you ask I can make a gh repo.
> 
> > the configuration object is effectively a configuration file that is specific to the apk. we tried to setup reasonable defaults, but android is a constantly evolving ecosystem...so those defaults might be wrong today.

Maybe this has to do with android intent filters.
> The permission is there but maybe it is the selecting of a subdir which is never prompted. I take it the app will have basic write permission then permission to a single dir.
> 
> > not sure why it is not asking for permission. you might be able to use the android app info to manually enable the permissions.
> 
> Yes, it is well to save the map data anyway.
> 
> > long story short, the permissions are used so that we can write a cache on the device of downloaded map tiles. If we dont cache them, the performance is awful and then you're constantly downloading tiles.

**Comment by spyhunter99** [comment]
The config class can be overridden by developers.  Osmdroid is more of a
developer library than an app. The app on the playstore is really just a
demo of what the library can do

**Comment by freedom-foundation** [comment]
Not sure why you are calling this cache. It is storing a map data file. Cache is usually a CPU or RAM buffer on a specific memory for such use, or a streaming data buffer. 
> long story short, the permissions are used so that we can write a cache on the device of downloaded map tiles. If we dont cache them, the performance is awful and then you're constantly downloading tiles.
